src/Middleware/Diagnostics/src/StatusCodePage/StatusCodePagesMiddleware.cs

```diff
@@ -48,8 +48,14 @@ public async Task Invoke(HttpContext context)
             statusCodeFeature.Enabled = false;
         }
 
+        // Attach pathFormat to HttpContext.Items early in the pipeline
+        context.Items[nameof(StatusCodePagesOptions)] = _options.PathFormat;
+
         await _next(context);
 
+        // Remove pathFormat from HttpContext.Items after handler execution
+        context.Items.Remove(nameof(StatusCodePagesOptions));
+
```

Comment:
Not suggesting we change this right now, but maybe this should go into https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/Diagnostics/src/StatusCodePage/StatusCodePagesFeature.cs instead.

The items dictionary is typically not used by the framework itself.